src/System.Linq/src/System/Linq/AppendPrepend.SpeedOpt.cs

```diff
@@ -181,19 +181,17 @@ public override List<TSource> ToList()
             {
                 int count = GetCount(onlyIfCheap: true);
                 List<TSource> list = count == -1 ? new List<TSource>() : new List<TSource>(count);
+
                 for (SingleLinkedNode<TSource> node = _prepended; node != null; node = node.Linked)
                 {
                     list.Add(node.Item);
                 }
 
                 list.AddRange(_source);
+
                 if (_appended != null)
                 {
-                    IEnumerator<TSource> e = _appended.GetEnumerator(_appendCount);
-                    while (e.MoveNext())
-                    {
-                        list.Add(e.Current);
-                    }
+                    list.AddRange(_appended.ToArray(_appendCount));
```

Comment:
If list is was created with default capacity, this change can make clearly performance improment. Well done!